src/proof_system/proof_system_errors.rs

```diff
@@ -6,17 +6,22 @@
 
 //! Errors related to the proof_system module.
 
-use thiserror::Error;
+use crate::commitment_scheme::kzg10::errors::KZG10Errors;
+use crate::constraint_system::cs_errors::PreProcessingErrors;
+use crate::fft::fft_errors::FFTErrors;
 
 /// Defines all of the possible ProofError types that we could have when
 /// we are working with the `proof_system` module.
-#[derive(Error, Debug)]
+#[derive(core::fmt::Debug)]
 pub enum ProofErrors {
     /// This error occurs when the verification of a `Proof` fails.
-    #[error("proof verification failed")]
     ProofVerificationError,
-    /// This error occurrs when the Prover structure already contains a
-    /// preprocessed circuit inside, but you call preprocess again.
-    #[error("circuit already preprocessed")]
-    CircuitAlreadyPreprocessed,
+    /// This error occurs when the evaluation domain can not be constructed.
+    CouldNotConstructEvaluationDomain(FFTErrors),
+    /// This error occurs when preprocessing of a circuit fails.
+    CouldNotPreProcessCircuit(PreProcessingErrors),
+    /// This error occurs when a poly can not be committed to.
+    CouldNotCommitToPoly(KZG10Errors),
+    /// This error occurs when the computation of the quotient poly fails.
+    CouldNotComputeQuotientPoly(FFTErrors),
```

Comment:
Instead of this wrappers, I would either have a centralized file containing all of the errors of the crate so that you don't need to make wrappers or either implement the corresponding conversion implementations (From trait impls).

Reply:
Good point. I may have missed this when you suggested it, cause I felt like this was quite an ugly way to do it.